tools/scripts/clone.js

```diff
@@ -0,0 +1,42 @@
+const fs = require('fs')
+const path = require('path')
+
+const buildDir = path.resolve(__dirname, '../../build/')
+const builtFileNames = fs.readdirSync(buildDir)
+
+async function run () {
+  const files = await Promise.all(builtFileNames.map(fileName => {
+    return fs.promises.readFile(`${buildDir}/${fileName}`, 'utf-8')
+  }))
+
+  const fuzzyMajor = await Promise.all(files.map((f, i) => {
+    const fileName = builtFileNames[i]
+    const content = f
+    if (fileName.startsWith('nr-loader') && fileName.endsWith('.js')) {
+      const pieces = fileName.split(/-(?=\d)/).map(x => x.split('.'))
+      if (!pieces[1]) return Promise.resolve()
+      pieces[1] = pieces[1].map((v, i) => i === 1 || i === 2 ? 'x' : v).join('.')
+      const major = pieces.join('-')
+      return fs.promises.writeFile(`${buildDir}/${major}`, content)
+    }
+    return Promise.resolve()
+  }))
+
+  const fuzzyMinor = await Promise.all(files.map((f, i) => {
+    const fileName = builtFileNames[i]
+    const content = f
+    if (fileName.startsWith('nr-loader') && fileName.endsWith('.js')) {
+      const pieces = fileName.split(/-(?=\d)/).map(x => x.split('.'))
+      if (!pieces[1]) return Promise.resolve()
+      pieces[1] = pieces[1].map((v, i) => i === 2 ? 'x' : v).join('.')
+      const minor = pieces.join('-')
+      return fs.promises.writeFile(`${buildDir}/${minor}`, content)
+    }
+    return Promise.resolve()
+  }))
```

Comment:
Bit confusing if someone had no idea what this was about.
```suggestion
  const fuzzyMajor = await Promise.all(files.map((f, i) => {
    const fileName = builtFileNames[i]
    // Assuming the filename contains a semantic version pattern, "-#.#.#.", replace the minor and patch numbers.
    const allMinorAndPatch = fileName.replace(/(^nr-loader.*-\d+\.)(\d+)\.(\d+)(.*\.js$)/, "$1x.x$4")
    if (allMinorAndPatch !== fileName) { // we only get the same string back if the filename didn't have that pattern, in which case we'll skip this
      return fs.promises.writeFile(`${buildDir}/${allMinorAndPatch}`, f)
    }
    return Promise.resolve()
  }))

  const fuzzyMinor = await Promise.all(files.map((f, i) => {
    const fileName = builtFileNames[i]
    // Assuming the filename contains a semantic version pattern, "-#.#.#.", replace the patch number.
    const allPatch = fileName.replace(/(^nr-loader.*-\d+\.\d+\.)(\d+)(.*\.js$)/, "$1x$3")
    if (allPatch !== fileName) { // we only get the same string back if the filename didn't have that pattern, in which case we'll skip this
      return fs.promises.writeFile(`${buildDir}/${allPatch}`, f)
    }
    return Promise.resolve()
  }))
```
![image](https://user-images.githubusercontent.com/28714891/233728376-e8028c0c-7d1c-4e50-9361-d735d3fe9075.png)
![image](https://user-images.githubusercontent.com/28714891/233728406-403429dd-fd7f-4f07-87fa-a45a01ec6dd0.png)